src/raft_log.rs

```diff
@@ -354,12 +354,13 @@ impl<T: Storage> RaftLog<T> {
         term > self.last_term() || (term == self.last_term() && last_index >= self.last_index())
     }
 
-    /// Returns any entries since the a particular index.
-    pub fn next_entries_since(&self, since_idx: u64) -> Option<Vec<Entry>> {
+    /// Returns any entries between max(`since_idx` + 1, first_index)
+    /// and min(`persisted_idx`, committed) + 1.
+    pub fn next_entries_between(&self, since_idx: u64, persisted_idx: u64) -> Option<Vec<Entry>> {
         let offset = cmp::max(since_idx + 1, self.first_index());
-        let committed = self.committed;
-        if committed + 1 > offset {
-            match self.slice(offset, committed + 1, None) {
+        let high = cmp::min(persisted_idx, self.committed) + 1;
```

Comment:
Could `persisted_idx ` be smaller than `self.committed`?

Reply:
Yes. If it is a follower, it may receive some committed entries and they are not persisted in local.